test/e2e/test_pubsub.go

```diff
@@ -24,120 +24,87 @@ import (
-	}()
-
-	if err := client.WaitForResourceReady(client.Namespace, psName, schema.GroupVersionResource{
-		Group:    "events.cloud.google.com",
-		Version:  "v1alpha1",
-		Resource: "pubsubs",
-	}); err != nil {
+	if err := client.Core.WaitForResourceReady(psName, lib.PubsubTypeMeta); err != nil {
 		t.Error(err)
 	}
 }
 
 // PubSubWithTargetTestImpl tests we can receive an event from PubSub. If assertMetrics is set to true, we also assert
 // for StackDriver metrics.
-func PubSubWithTargetTestImpl(t *testing.T, packages map[string]string, assertMetrics bool) {
-	topicName, deleteTopic := makeTopicOrDie(t)
+func PubSubWithTargetTestImpl(t *testing.T, assertMetrics bool) {
+	topicName, deleteTopic := lib.MakeTopicOrDie(t)
 	defer deleteTopic()
 
 	psName := helpers.AppendRandomString(topicName + "-pubsub")
 	targetName := helpers.AppendRandomString(topicName + "-target")
 
-	client := Setup(t, true)
+	client := lib.Setup(t, true)
 	if assertMetrics {
 		client.SetupStackDriverMetrics(t)
 	}
-	defer TearDown(client)
-
-	config := map[string]string{
-		"namespace":  client.Namespace,
-		"topic":      topicName,
-		"pubsub":     psName,
-		"targetName": targetName,
-		"targetUID":  uuid.New().String(),
-	}
-	for k, v := range packages {
-		config[k] = v
-	}
-
-	installer := NewInstaller(client.Dynamic, config,
-		EndToEndConfigYaml([]string{"pubsub_target", "istio"})...)
-
-	// Create the resources for the test.
-	if err := installer.Do("create"); err != nil {
-		t.Errorf("failed to create, %s", err)
-		return
-	}
-
-	// Delete deferred.
-	defer func() {
-		if err := installer.Do("delete"); err != nil {
-			t.Errorf("failed to create, %s", err)
-		}
-		// Just chill for tick.
-		time.Sleep(10 * time.Second)
-	}()
-
-	gvr := schema.GroupVersionResource{
-		Group:    "events.cloud.google.com",
-		Version:  "v1alpha1",
-		Resource: "pubsubs",
-	}
-
-	jobGVR := schema.GroupVersionResource{
-		Group:    "batch",
-		Version:  "v1",
-		Resource: "jobs",
-	}
-
-	if err := client.WaitForResourceReady(client.Namespace, psName, gvr); err != nil {
+	defer lib.TearDown(client)
+
+	// Create a target Job to receive the events.
+	job := resources.TargetJob(targetName, []v1.EnvVar{{
+		Name:  "TARGET",
+		Value: "falldown",
+	}})
+	client.CreateJobOrFail(job, lib.WithServiceForJob(targetName))
+
+	// Create the PubSub source.
+	eventsPubsub := kngcptesting.NewPubSub(psName, client.Namespace,
+		kngcptesting.WithPubSubSink(metav1.GroupVersionKind{
+			Version: "v1",
+			Kind:    "Service"}, targetName),
+		kngcptesting.WithPubSubTopic(topicName))
+	client.CreatePubSubOrFail(eventsPubsub)
+
+	// Create the Istio ServiceEntry.
+	istioServiceEntry := resources.IstioServiceEntry(
+		"cloud-run-events-googleapis-ext", client.Namespace)
+	client.CreateUnstructuredObjOrFail(istioServiceEntry)
+
+	if err := client.Core.WaitForResourceReady(psName, lib.PubsubTypeMeta); err != nil {
 		t.Error(err)
 	}
 
-	topic := getTopic(t, topicName)
+	time.Sleep(1 * time.Minute)
```

Comment:
1 min? Seems a lot? Maybe add a comment saying why...

Reply:
This is ported from the original test. But I tried to run the test without it, the test can still pass.
So removed it.